third-party/py/pex/pex/compiler.py

```diff
@@ -58,6 +60,17 @@ def main(root, relpaths):
 """
 
 
+@contextlib.contextmanager
+def named_temporary_file():
+    fp, path = tempfile.mkstemp()
+    os.close(fp)
+    try:
+        with open(path, 'w') as fp:
+            yield fp
+    finally:
+        os.remove(path)
```

Comment:
I'll have to try this when I get into the office, but when I was using something like, this, I was getting an exception with the call to `remove` because the `subprocess` call hadn't finished closing it yet.